src/codemodder/codemods/sonar.py

```diff
@@ -54,20 +54,21 @@ def from_core_codemod(
 class SonarDetector(BaseDetector):
     _lazy_cache = None
 
-    def _process_sonar_findings(self, sonar_json_files: list[str]) -> SonarResultSet:
-        combined_result_set = SonarResultSet()
-        for file in sonar_json_files or []:
-            combined_result_set |= SonarResultSet.from_json(file)
-        return combined_result_set
-
     def apply(
         self,
         codemod_id: str,
         context: CodemodExecutionContext,
         files_to_analyze: list[Path],
     ) -> ResultSet:
         if not self._lazy_cache:
-            self._lazy_cache = self._process_sonar_findings(
+            self._lazy_cache = process_sonar_findings(
                 context.tool_result_files_map.get("sonar", [])
             )
         return self._lazy_cache
+
+
+def process_sonar_findings(sonar_json_files: list[str]) -> SonarResultSet:
```

Comment:
We could probably just use `@functools.cache` here and get rid of the `_lazy_cache` on the class?

Reply:
I swear I recall there was a reason this was written as is, I think @andrecsilva wrote it originally? because we also posed using @functools.cache at that time

Reply:
going to merge so I can use this PR to add more sonar codemods, but can come back to this if needed